test/integration/graql/query/GraqlDeleteIT.java

```diff
@@ -110,527 +110,8 @@ public static void closeSession() {
-
-        assertExists(tx, var().id(ownerId.toString()));
-        assertExists(tx, var().id(attrId.toString()));
-        assertNotExists(tx, Graql.and(var().id(ownerId.toString()).has("title", "a"), var("a").id(attrId.toString())));
-
-        // using `thing` throws
-        answers = tx.execute(Graql.parse("match $x isa thing, has title $a; get; limit 1;").asGet());
-        ownerId = answers.get(0).get("x").id();
-        attrId = answers.get(0).get("a").id();
-
-        exception.expect(GraqlSemanticException.class);
-        exception.expectMessage("Cannot delete attribute ownership, concept [$a]");
-        exception.expectMessage("is not of required attribute type [thing]");
-        tx.execute(Graql.parse("match $x id " + ownerId + "; $x has title $a; $a id " + attrId + "; delete $x has thing $a;").asDelete());
-    }
-
-    @Test
-    public void whenDeletingAttributeOwnershipWithSubtype_Throw() {
-        AttributeType<String> firstName = tx.putAttributeType("first-name", AttributeType.ValueType.STRING);
-        firstName.sup(tx.getAttributeType("name"));
-        tx.getEntityType("person").has(firstName);
-
-        List<ConceptMap> answers = tx.execute(Graql.parse("match $x isa person, has name $n; get;").asGet());
-        assertEquals(10, answers.size());
-        tx.execute(Graql.parse("insert $x isa person, has first-name \"john\";").asInsert());
-
-        answers = tx.execute(Graql.parse("match $x isa person, has name $n; get;").asGet());
-        assertEquals(11, answers.size());
-
-        // `first-name` will not be satisfied by all `name` attributes
-        exception.expect(GraqlSemanticException.class);
-        exception.expectMessage("Cannot delete attribute ownership, concept [$n]");
-        exception.expectMessage("is not of required attribute type [first-name]");
-        tx.execute(Graql.parse("match $x isa person, has name $n; delete $x has first-name $n;").asDelete());
-    }
-
-
-    @Test
-    public void matchInstance_DeleteWithSupertype() {
-        List<ConceptMap> answers = tx.execute(Graql.parse("match $m isa movie, has title \"Godfather\"; get;").asGet());
-        assertEquals(1, answers.size());
-        tx.execute(Graql.parse("match $m isa movie, has title \"Godfather\"; delete $m isa entity;").asDelete());
-        answers = tx.execute(Graql.parse("match $m isa movie, has title \"Godfather\"; get;").asGet());
-        assertEquals(0, answers.size());
-
-        answers = tx.execute(Graql.parse("match $m isa movie, has title \"The Muppets\"; get;").asGet());
-        assertEquals(1, answers.size());
-        tx.execute(Graql.parse("match $m isa movie, has title \"The Muppets\"; delete $m isa thing;").asDelete());
-        answers = tx.execute(Graql.parse("match $m isa movie, has title \"The Muppets\"; get;").asGet());
-        assertEquals(0, answers.size());
-    }
-
-
-    @Test
-    public void deleteWithDirectType_ThrowsWhenNotDirectType() {
```

Comment:
we have the BDD test for direct type succeeding, do we have this negative case as well ?

Reply:
Yes